python/lsst/ts/attcpip/csc.py

```diff
@@ -0,0 +1,292 @@
+
+        # TODO DM-38912 Get this from the configuration.
+        host = ""
+        cmd_evt_port = 0
+        telemetry_port = 0
+
+        if self.simulation_mode == 1:
+            assert self.simulator is not None
+            await self.simulator.cmd_evt_server.start_task
+            await self.simulator.telemetry_server.start_task
+            host = self.simulator.cmd_evt_server.host
+            cmd_evt_port = self.simulator.cmd_evt_server.port
+            telemetry_port = self.simulator.telemetry_server.port
+
+        self.cmd_evt_client = tcpip.Client(
+            host=host, port=cmd_evt_port, log=self.log, name="CmdEvtClient"
+        )
+        self.telemetry_client = tcpip.Client(
+            host=host, port=telemetry_port, log=self.log, name="TelemetryClient"
+        )
+
+        await self.cmd_evt_client.start_task
+        await self.telemetry_client.start_task
+
+        self._event_task = asyncio.create_task(self.cmd_evt_loop())
+        self._telemetry_task = asyncio.create_task(self.telemetry_loop())
+
+    async def stop_clients(self) -> None:
+        if not self.connected:
+            self.log.warning("Not connected. Ignoring.")
+            return
+
+        self._event_task.cancel()
+        self._telemetry_task.cancel()
+
+        await self.cmd_evt_client.close()
+        await self.telemetry_client.close()
+
+        if self.simulation_mode == 1 and self.simulator:
+            await self.simulator.telemetry_server.close()
+            await self.simulator.cmd_evt_server.close()
+
+    async def close_tasks(self) -> None:
+        """Shut down pending tasks. Called by `close`.
+
+        Perform all cleanup other than disabling logging to SAL
+        and closing the dds domain.
+        """
+        await self.stop_clients()
+        await super().close_tasks()
+
+    async def cmd_evt_loop(self) -> None:
+        """Execute the command and event loop.
+
+        This loop waits for incoming command and event messages and processes
+        them when they arrive.
+        """
+        while True:
+            data = await self.cmd_evt_client.read_json()
+            data_id: str = data["id"]
+
+            # If data_id starts with "evt_" then handle the event data.
+            if data_id.startswith("evt_"):
+                await self.call_set_write(data=data)
+            elif CommonCommandArgument.SEQUENCE_ID in data:
+                sequence_id = data[CommonCommandArgument.SEQUENCE_ID]
+                response = data[CommonCommandArgument.ID]
+                match response:
+                    case Ack.ACK:
+                        self.commands_issued[sequence_id].set_ack()
+                    case Ack.NOACK:
+                        self.commands_issued[sequence_id].set_noack()
+                    case Ack.SUCCESS:
+                        self.commands_issued[sequence_id].set_success()
+                    case Ack.FAIL:
+                        self.commands_issued[sequence_id].set_fail()
```

Comment:
I thought we will usually add the `default` case for `switch`. In Python, it would be `case _:`:
https://www.freecodecamp.org/news/python-switch-statement-switch-case-example/

Reply:
Good point! Default case added.